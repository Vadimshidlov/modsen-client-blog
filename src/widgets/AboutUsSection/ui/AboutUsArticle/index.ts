export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
