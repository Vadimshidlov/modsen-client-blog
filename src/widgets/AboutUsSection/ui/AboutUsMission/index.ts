export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
