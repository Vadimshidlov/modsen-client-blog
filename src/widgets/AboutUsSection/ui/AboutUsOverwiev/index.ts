export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
