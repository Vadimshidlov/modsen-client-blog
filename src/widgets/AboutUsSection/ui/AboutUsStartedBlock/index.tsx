export { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock/AboutUsStartedBlock';
