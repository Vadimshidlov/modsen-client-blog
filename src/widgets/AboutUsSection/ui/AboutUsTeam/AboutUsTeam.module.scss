.aboutUsTeamSection {
  display: flex;
  justify-content: space-between;
  align-items: center;
  column-gap: 80px;
  margin-bottom: 128px;

  @media (max-width: $tablet) {
    margin-bottom: 64px;
  }

  @media (max-width: $mobile) {
    margin-bottom: 32px;
  }

  @media (max-width: 1000px) {
    flex-direction: column-reverse;
    row-gap: 20px;
  }

  .aboutUsTeamImageWrapper {
    position: relative;

    &::after {
      z-index: 2;
      content: '';
      position: absolute;
      display: block;
      top: 30%;
      left: -50px;
      width: 103px;
      height: 116px;
      border-top-left-radius: 50%;
      background-color: $yellow;

      @media (max-width: $tablet) {
        top: 45%;
        left: -24px;
        width: 51px;
        height: 56px;
      }

      @media (max-width: $mobile) {
        top: 45%;
        left: -12px;
        width: 26px;
        height: 28px;
      }
    }

    .aboutUsTeamImage {
      width: 100%;
      object-fit: cover;

      @media (max-width: $mobile) {
        height: 305px;
      }
    }
  }
}
