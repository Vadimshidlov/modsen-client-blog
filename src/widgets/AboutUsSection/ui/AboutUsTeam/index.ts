export { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam/AboutUsTeam';
