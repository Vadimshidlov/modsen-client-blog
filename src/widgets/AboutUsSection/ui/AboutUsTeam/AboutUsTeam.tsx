import React from 'react';
import Image from 'next/image';
import styles from '@/widgets/AboutUsSection/ui/AboutUsTeam/AboutUsTeam.module.scss';
import aboutUsHandsImageSrc from '@/widgets/AboutUsSection/assets/aboutUsHands.png';
import { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle';

export function AboutUsTeam() {
  const title = 'Our team of creatives';
  const subTitle = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.';
  const text = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
  ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
  ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
  reprehenderit in voluptate velit esse cillum dolore eu fugiat.`;

  return (
    <section className={styles.aboutUsTeamSection}>
      {/* <article className={styles.aboutUsTextContent}>
        <h2 className={styles.aboutUsTeamTitle}>Our team of creatives</h2>
        <h4 className={styles.aboutUsTeamSubtitle}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.
        </h4>
        <span className={styles.aboutUsText}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
          ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
          ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
          reprehenderit in voluptate velit esse cillum dolore eu fugiat.
        </span>
      </article> */}
      <AboutUsArticle title={title} subTitle={subTitle} text={text} />
      <div className={styles.aboutUsTeamImageWrapper}>
        <Image
          placeholder="blur"
          className={styles.aboutUsTeamImage}
          src={aboutUsHandsImageSrc}
          alt="author-avatar"
        />
      </div>
    </section>
  );
}
