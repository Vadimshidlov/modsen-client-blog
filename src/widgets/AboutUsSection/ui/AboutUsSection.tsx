import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsSection.module.scss';
import container from '@/shared/styles/container.module.scss';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';
import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';
import { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock';

export function AboutUsSection() {
  return (
    <section className={`${styles.aboutUsSection} ${container.container}`}>
      <AboutUsTitle />
      <AboutUsOverwiev />
      <AboutUsMission />
      <AboutUsTeam />
      <AboutUsStartedBlock />
    </section>
  );
}
