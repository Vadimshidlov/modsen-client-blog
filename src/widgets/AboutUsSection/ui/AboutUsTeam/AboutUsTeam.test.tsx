import { render } from '@testing-library/react';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';

jest.mock('next/font/google', () => ({
  Inter: () => ({ variable: 'fake-variable' }),
}));

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AboutUsTeam tests', () => {
  test('renders article section with correct data', () => {
    const { getByText } = render(<AboutUsTeam />);

    expect(getByText('Mocked title')).toBeInTheDocument();
    expect(getByText('Mocked subTitle')).toBeInTheDocument();
  });

  test('renders image with correct alt text', () => {
    const { getByAltText } = render(<AboutUsTeam />);

    expect(getByAltText('author-avatar')).toBeInTheDocument();
  });

  test('should matches snapshot', () => {
    const { container } = render(<AboutUsTeam />);
    expect(container).toMatchSnapshot();
  });
});
