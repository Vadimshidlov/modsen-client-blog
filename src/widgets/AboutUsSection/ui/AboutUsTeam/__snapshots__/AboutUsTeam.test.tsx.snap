// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsTeam tests should matches snapshot 1`] = `
<div>
  <section
    class="aboutUsTeamSection"
  >
    <article
      class="aboutUsTextContent"
    >
      <h2
        class="aboutUsTeamTitle"
      >
        Mocked title
      </h2>
      <h4
        class="aboutUsTeamSubtitle"
      >
        Mocked subTitle
      </h4>
      <span
        class="aboutUsText"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
  ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
  ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
  reprehenderit in voluptate velit esse cillum dolore eu fugiat.
      </span>
    </article>
    <div
      class="aboutUsTeamImageWrapper"
    >
      <img
        alt="author-avatar"
        class="aboutUsTeamImage"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        width="40"
      />
    </div>
  </section>
</div>
`;
